File: modules/x-engine-jsi-ui/android/jsi-ui/src/main/java/com/zkty/jsi/JSI_ui.java
package com.zkty.jsi;

import androidx.annotation.Nullable;

import com.zkty.engine.nativ.protocol.IUI;
import com.zkty.nativ.core.NativeContext;
import com.zkty.nativ.core.NativeModule;
import com.zkty.nativ.jsi.bridge.CompletionHandler;
import com.zkty.nativ.ui.NativeUI;


public class JSI_ui extends xengine_jsi_ui{
    private NativeUI iUi;
    @Override
    protected void afterAllJSIModuleInited() {
        NativeModule module = NativeContext.sharedInstance().getModuleByProtocol(IUI.class);
        if (module instanceof NativeUI)
            iUi = (NativeUI) module;
    }

    @Override
    public void _setNavTitle(NavTitleDTO dto, CompletionHandler<Nullable> handler) {
        iUi.setNavTitle(dto.title,dto.titleColor,dto.titleSize);
    }

    @Override
    public void _setNavBarHidden(NavHiddenBarDTO dto, CompletionHandler<Nullable> handler) {
        iUi.setNavBarHidden(dto.isHidden,dto.isAnimation);
    }


}

File: modules/x-engine-jsi-ui/iOS/Class/JSI_ui.m
//
//  JSI_ui.m
//  ui
//
//  Created by zk on 2020/9/7.
//  Copyright © 2020 edz. All rights reserved.


#import "JSI_ui.h"
#import "JSIContext.h"
#import "XENativeContext.h"
#import "iUI.h"

@interface JSI_ui ()
@property (nonatomic, strong) id<iUI> ui;
@end

@implementation JSI_ui
JSI_MODULE(JSI_ui)
  
- (void)afterAllJSIModuleInited {
    self.ui = [[XENativeContext sharedInstance] getModuleByProtocol:@protocol(iUI)];
}

-(void)_setNavBarHidden:(NavHiddenBarDTO *)dto complete:(void (^)(BOOL))completionHandler{
    [self.ui setNavBarHidden:dto.isHidden isAnimation:dto.isAnimation];
    completionHandler(YES);
}


- (void)_setNavTitle:(NavTitleDTO *)dto complete:(void (^)(BOOL))completionHandler {
    [self.ui setNavTitle:dto.title withTitleColor:dto.titleColor withTitleSize:dto.titleSize];
    completionHandler(YES);
}

@end
 
